dan_layer/engine/tests/common/src/lib.rs

```diff
@@ -81,11 +85,10 @@ unsafe extern "C" fn tari_alloc(len: u32) -> *mut u8 {
     ptr
 }
 
-#[no_mangle]
-unsafe extern "C" fn tari_free(ptr: *mut u8) {
+pub unsafe fn tari_free(ptr: *mut u8) {
```

Comment:
If `common::tari_alloc` are `extern`s they do not need to be "proxied" in the macro-generated code.